remote-messaging/remote-messaging-impl/src/main/java/com/duckduckgo/remote/messaging/impl/models/MatchingAttributes.kt

```diff
@@ -0,0 +1,186 @@
+    ) : MatchingAttribute(), BooleanMatchingAttribute
+
+    data class WidgetAdded(
+        override val value: Boolean,
+        val fallback: Boolean? = null
+    ) : MatchingAttribute(), BooleanMatchingAttribute
+
+    data class SearchCount(
+        override val min: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        override val max: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        override val value: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        val fallback: Boolean? = null
+    ) : MatchingAttribute(), RangeIntMatchingAttribute, IntMatchingAttribute
+
+    data class Bookmarks(
+        override val min: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        override val max: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        override val value: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        val fallback: Boolean? = null
+    ) : MatchingAttribute(), RangeIntMatchingAttribute, IntMatchingAttribute
+
+    data class Favorites(
+        override val min: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        override val max: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        override val value: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        val fallback: Boolean? = null
+    ) : MatchingAttribute(), RangeIntMatchingAttribute, IntMatchingAttribute
+
+    data class AppTheme(
+        override val value: String = MATCHING_ATTR_STRING_DEFAULT_VALUE,
+        val fallback: Boolean? = null
+    ) : MatchingAttribute(), StringMatchingAttribute
+
+    data class DaysSinceInstalled(
+        override val min: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        override val max: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        override val value: Int = MATCHING_ATTR_INT_DEFAULT_VALUE,
+        val fallback: Boolean? = null
+    ) : MatchingAttribute(), RangeIntMatchingAttribute, IntMatchingAttribute
+
+    data class DaysUsedSince(
+        override val since: Date,
+        override val value: Int,
+        val fallback: Boolean? = null
+    ) : MatchingAttribute(), DateMatchingAttribute
+
+    data class Unknown(val fallback: Boolean?) : MatchingAttribute()
+}
+
+interface RangeIntMatchingAttribute {
+    val min: Int
+    val max: Int
+}
+
+interface RangeStringMatchingAttribute {
+    val min: String
+    val max: String
+}
+
+interface BooleanMatchingAttribute {
+    val value: Boolean
+}
+
+interface StringMatchingAttribute {
+    val value: String
+}
+
+interface IntMatchingAttribute {
+    val value: Int
+}
+
+interface StringArrayMatchingAttribute {
+    val value: List<String>
+}
+
+interface DateMatchingAttribute {
+    val since: Date
+    val value: Int
+}
+
+@Suppress("UNCHECKED_CAST")
+fun Any?.toStringList(): List<String> = this?.let { it as List<String> } ?: emptyList()
+
+fun Any?.toIntOrDefault(default: Int): Int = when {
+    this == null -> default
+    this is Double -> this.toInt()
+    this is Long -> this.toInt()
+    else -> this as Int
+}
+
+fun Any?.toStringOrDefault(default: String): String = this?.let { it as String } ?: default
```

Comment:
This extension `fun`s should definitely be marked as `internal`. We don't want extension funs on `Any` visible outside this module